packages/block-library/src/table/index.js

```diff
@@ -111,17 +129,31 @@ export const settings = {
 	edit,
 
 	save( { attributes } ) {
-		const { hasFixedLayout, head, body, foot } = attributes;
+		const {
+			className,
+			hasFixedLayout,
+			head,
+			body,
+			foot,
+			backgroundColor,
+			customBackgroundColor,
+			textColor,
+			customTextColor,
+		} = attributes;
 		const isEmpty = ! head.length && ! body.length && ! foot.length;
 
 		if ( isEmpty ) {
 			return null;
 		}
 
-		const classes = classnames( {
+		const tableClasses = classnames( {
 			'has-fixed-layout': hasFixedLayout,
 		} );
 
+		const isStriped = !! className && className.indexOf( 'is-style-stripes' ) > -1;
```

Comment:
Not a fan of having to detect the style variation like this, I wonder if there's a better way?

Reply:
@talldan Use `className.includes( 'is-style-stripes' )`.

Reply:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/includes

Reply:
@ZebulanStanphill Cheers for the suggestion, though I was referring to checking for values in strings being a brittle implementation rather than the method used to check them - see http://wiki.c2.com/?StringlyTyped.

Reply:
Ah, I see.

Perhaps there should be a `getStyleClass` function defined somewhere that you could pass the style name into, which would return the class for that style, abstracting away the `is-style-` part.

Or perhaps there should be a `getBlockStyle` function that determines if a block is using a style variation somehow. Of course, the only way to check right now is via classes, but if it was later decided to include style variation names in the comment delimiters or something like that, only that function would have to be changed.

Reply:
I've changed the implementation now to use a different approach and this code no longer exists, so lets not worry about it for now 😄